pkg/controller/atlas/provider.go

```diff
@@ -81,9 +81,10 @@ func (p *ProductionProvider) IsResourceSupported(resource akov2.AtlasCustomResou
 		*akov2.AtlasStreamConnection:
 		return false
 	case *akov2.AtlasDeployment:
-		if atlasResource.Spec.ServerlessSpec == nil {
-			return true
-		}
+		hasSearchNodes := atlasResource.Spec.DeploymentSpec != nil && len(atlasResource.Spec.DeploymentSpec.SearchNodes) >= 0
```

Comment:
```suggestion
		hasSearchNodes := atlasResource.Spec.DeploymentSpec != nil && len(atlasResource.Spec.DeploymentSpec.SearchNodes) > 0
```

if `len(atlasResource.Spec.DeploymentSpec.SearchNodes) == 0`, then no search nodes are defined.